content/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -178,177 +80,24 @@ of the altered protocol.
+## Peer Discovery
 
-- While in this state:
-  - `ChainSync` is well-bootstrapped, and has an initial **trusted** `StateTree` to start from.
-  - `ChainSync` is receiving latest `Blocks` from `BlockPubsub`
-  - `ChainSync` starts fetching and validating blocks
-  - `ChainSync` has unvalidated blocks between `ChainSync.FinalityTipset` and `ChainSync.TargetHeads`
-- **Chain State and Finality**:
-  - In this state, the **chain MUST NOT advance** beyond whatever the node already has:
-    - `FinalityTipset` does not change.
-    - No new blocks are reported to consumers/users of `ChainSync` yet.
-    - The chain state provided is the available `Blocks` and `StateTree` for all available epochs,
-      specially the `FinalityTipset`.
-    - Finality must not move forward here because there are serious attack vectors where a node can be forced to end up on the wrong fork if finality advances before validation is complete up to the block production fringe.
-  - Validation must advance, all the way to the block production fringe:
-    - Validate the whole chain, from `FinalityTipset` to `BestTargetHead`
-    - The node can reach `BestTargetHead` only to find out it was invalid, then has to update `BestTargetHead` with next best one, and sync to it
-      (without having advanced `FinalityTipset` yet, as otherwise we may end up on the wrong fork)
-- **security conditions to transition out:**
-  - Gaps between `ChainSync.FinalityTipset ... ChainSync.BestTargetHead` have been closed:
-    - All `Blocks` and their content MUST be fetched, stored, linked, and validated locally.
-      This includes `BlockHeaders`, `Messages`, etc.
-    - Bad heads have been expunged from `ChainSync.TargetHeads`. Bad heads include heads that initially
-      seemed good but turned out invalid, or heads that `ChainSync` has failed to connect (ie. cannot
-      fetch ancestors connecting back to `ChainSync.FinalityTipset` within a reasonable amount of time).
-    - All blocks between `ChainSync.FinalityTipset ... ChainSync.TargetHeads` have been validated
-      This means all blocks _before_ the best heads.
-  - Not under a temporary network partition
-- **transitions out:**
-  - once gaps between `ChainSync.FinalityTipset ... ChainSync.TargetHeads` are closed: move to `CHAIN_FOLLOW`
-  - (Perhaps moving to `CHAIN_FOLLOW` when 1-2 blocks back in validation may be ok.
-    - we dont know we have the right head until we validate it, so if other heads of similar height are right/better, we won't know until then.)
+Peer discovery is a critical part of the overall architecture. Taking this wrong can have severe consequences for the operation of the protocol. This is especially important for security because network "Bootstrap" is a difficult problem in peer-to-peer networks. The set of peers a new node initially connects to may completely dominate the node's awareness of other peers, and therefore the view of the state of the network that the node has.
```

Comment:
> Taking this wrong can have severe consequences for the operation of the protocol.
What does this mean?

Other than bootstrap nodes (hardcoded in the implementation) there is no wrong or right peer, we just can't assume anything and verify everything.

Reply:
Agree that it's important, let's make it a bit more explicit if possible. Instead of taking about 'peer discovery' in general and mentioning the bootstrap nodes as a particular case down the road (when it's actually the only case where it's important) I would just say something more direct as you just did now in your response.